### PR TITLE
Refactored netcdf datasource

### PR DESCRIPTION
**Pull request by jackieleng**
Made things more consistent in NetcdfDataSource. The following methods are now available:

```
+ get_values_by_ids (a list of ids)
+ get_values_by_id (convenience method if you want just one id)
+ get_values_by_timestep_nr (by timestep index)
```

Furthermore, `get_timeseries` now only accepts one variable and not a list of variables anymore.

### CODE REVIEW

File: views/map_animator.py [hunk 1]
```diff
@@ -209,9 +209,9 @@ def update_results(self):
 
             provider = layer.dataProvider()
 
-            values = ds.get_values_timestamp(parameter[0], timestamp_nr)
+            values = ds.get_values_by_timestamp(parameter[0], timestamp_nr)
```
Comment (bastiaan-roos) [inline views/map_animator.py]: This part is not working (existing bug), should be parameter instead of parameter[0]

File: views/map_animator.py [hunk 2]
```diff
@@ -209,9 +209,9 @@ def update_results(self):
...
             if stat == 'diff':
-                values = values - ds.get_values_timestamp(parameter[0], 0)
+                values = values - ds.get_values_by_timestamp(parameter[0], 0)
```
Comment (bastiaan-roos) [inline views/map_animator.py]: same as previous comment

File: datasource/netcdf.py [hunk 3]
```diff
@@ -158,35 +158,33 @@ def get_available_parameters(object_type):
 PUMPLIKE_OBJECTS = ['pumpstation', 'pumpline']
 
 
-def get_variables(object_type=None, parameters=[]):
+# TODO: use this for pumps still? Currently unused.
+def get_variable(object_type=None, parameter=None):
```
Comment (bastiaan-roos) [inline datasource/netcdf.py]: Throughout the code we use variable and parameter, better select one (variable?) and use this consequently

File: datasource/netcdf.py [hunk 4]
```diff
@@ -551,142 +549,126 @@ def obj_to_netcdf_id(self, object_id, normalized_object_type):
...
+        """Convenience method to get a regular array if only one ID is needed.
+        """
+        vals = self.get_values_by_ids(variable=variable,
+                                      object_type=object_type,
+                                      object_ids=[object_id],
+                                      caching=caching)
+        # Convert row array to regular array.
+        return vals[:, 0]
+
+    def get_values_by_timestamp(self, variable, timestamp_idx):
```
Comment (bastiaan-roos) [inline datasource/netcdf.py]: Better make naming around timestep en timestamp also consistent. Suggestion:
timestep_nr = nr of output timestep starting at 0 (is equal to index in netCDF)
timestep = time in seconds between the results
timestamp = time in seconds from start of the simulation, so timestep_nr \* timestep
datetime = seconds from 1-1-1970, is start_time + timestamp

**Comment by bastiaan-roos** [comment]
Why not clean the code further by removing all 'empty' functions (only with 'pass')
Line 84 to 110 of the netcdf.py file is not used outside the test (this is old code). remove code + tests?

**Comment by jackieleng** [comment]
Lines 84-110 might be useful if we decide to make the graph dropdown menu variables dynamic based on the currently active layer. E.g. for pumps only q_pump is relevant, but still all variables are shown now.

**Comment by jackieleng** [comment]
@bastiaan-roos Commentaar verwerkt.

**Pull request merged**